core/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/ExtendedBounds.java

```diff
@@ -166,8 +166,9 @@ ExtendedBounds parseAndValidate(String aggName, SearchContext context, DocValueF
         return new ExtendedBounds(min, max, minAsStr, maxAsStr);
     }
 
-    ExtendedBounds round(Rounding rounding) {
-        return new ExtendedBounds(min != null ? rounding.round(min) : null, max != null ? rounding.round(max) : null);
+    ExtendedBounds round(Rounding rounding, long offset) {
+        return new ExtendedBounds(min != null ? rounding.round(min) + offset : null,
+                max != null ? rounding.round(max) + offset : null);
```

Comment:
I don't think so. The min/max values should initially be the ones defined by the user in the query (e.g. "2016-01-01T04:00:00Z"). I we subtract the offset before rounding and e.g. do 1day intervals, we would get to 2015-12-31 in the example above, which I think is not the intention of the user. Does this make sense or am I missing something?